scripts/calculateRevenue.ts

```diff
@@ -23,13 +26,21 @@ async function main(args: ReturnType<typeof parseArgs>) {
   }> = []
 
   for (let i = 0; i < eligibleUsers.length; i++) {
-    const { referrerId, userAddress } = eligibleUsers[i]
+    const { referrerId, userAddress, timestamp } = eligibleUsers[i]
     console.log(
       `Calculating revenue for ${userAddress} (${i + 1}/${eligibleUsers.length})`,
     )
+
+    const referralDate = new Date(timestamp - REFERRAL_TIME_BUFFER_IN_SECONDS)
+    const periodStartDate = new Date(args['start-timestamp'])
+
     const revenue = await handler({
       address: userAddress,
-      startTimestamp: new Date(args['start-timestamp']),
+      // if the referral happened after the start of the period, only calculate revenue from the referral block onwards so that we exclude user activity before the referral
+      startTimestamp:
+        referralDate.getTime() > periodStartDate.getTime()
+          ? referralDate
+          : periodStartDate,
       endTimestamp: new Date(args['end-timestamp']),
```

Comment:
The updated code I'm working ensure startTimestamp < endTimestamp and throws if that's not the case.
What do you think of skipping early if `referralDate.getTime() >= endTimestamp`? Possibly logging something if useful.
And avoid the future exception.